accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,211 @@
+  For larger deployments there may be problems with the number of terminals or with network outages.
+  The SUMA Server must be reachable in order to perform any change of Terminal configuration.
+
+  ##### Terminal
+
+  - deployed from OS Image during boot (by applying Saltboot state)
+  - then works as standard SUMA minion
+
+
+  ##### Branch Server (alternative 1)
+
+  - SUMA Proxy
+  - downloads Images from SUMA (over HTTP) and distributes them to Terminals (over HTTP, TFTP, etc.)
+  - runs services needed by Terminals (DHCP, TFTP, HTTP, ...). The services are configured via Formulas with Forms.
+
+
+  ##### [Containerized Branch Server](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00087-containerized-branch.md) (alternative 2)
+
+  - SUMA Proxy
+  - TFTP redirected to HTTP
+  - Images downloaded via Squid HTTP proxy
+  - PXE entries are handled by Cobbler on SUMA Server
+  - no salt formulas, DHCP must be configured manually
+
+
+  ##### SUMA Server
+
+  - standard SUMA operations
+  - holds the configuration for Branch server services in Formulas
+  - controls boot process of each Terminal by applying Saltboot state
+  - triggers update of PXE data on Branch Server
+
+
+### 2. Full
+```
+    Terminal -> Branch Server -> Peripheral Server -> Hub
+```
+  Adding Hub on top of previous configuration solves the problems with scalability and offline operations.
+  It is suitable for for large organization with many department stores. It can look like this:
+
+  - Branch server on each floor
+  - Peripheral server in each building
+  - Hub in datacenter
+
+
+### 3. Combined Peripheral and Branch server
+```
+    Terminal -> Branch + Peripheral Server Combined -> Hub
+```
+
+  Customers with many small shops do need Hub for offline operations but do not need
+  separate Branch server and Peripheral Server because the number of connected terminals is low.
+  Running Branch + Peripheral Server on one machine could save resources.
+  The Proxy part of Branch server is not needed then.
+
+
+### 4. Single server
+```
+    Terminal -> Branch + SUMA Server Combined
+```
+  Suitable for one small shop.
+  This scenario is not a hard requirement but there were such requests for SLEPOS
+  so it would be nice to have it too.
+
+
+## Global Retail data
+
+The following items are typically shared in whole organization, so they should be visible and
+manageable via Hub.
+
+  - Images - distributed as large tarballs via http
+  - Image metadata - pillars, activation keys, list of repos and packages (in DB)
+  - [HWTYPE](https://www.uyuni-project.org/uyuni-docs/en/uyuni/retail/retail-deploy-terminals.html#_create_a_hardware_type_group) configurations
+
+## Local data
+
+The following items do not need central management and can stay on Peripheral Server (like in the current implementation).
+
+  - Branch groups -  groups containing Branch Server and connected terminals, includes Saltboot groups for Containerized Branch Server
```

Comment:
This Banch groups are normal systems groups, right? This can be managed from HUB server with the configuration formula.

Reply:
They are normal groups with attached group formulas. AFAIK the HUB configuration formula can't handle group formulas yet.

Reply:
correct, it can only create system groups.

Reply:
@rjmateus so is it OK to extend the HUB configuration formula to create also group formulas? It is the alternative 2 here https://github.com/nadvornik/uyuni-rfc/blob/hub/accepted/0000-hub_retail.md#hwtype-configurations

Reply:
from my point of view yes. In this way, we can have a centralized place to configure the peripheral servers.

Reply:
I have updated the RFC accordingly.